pkg/destroy/aws/shared.go

```diff
@@ -218,21 +217,22 @@ func (o *ClusterUninstaller) cleanSharedHostedZone(ctx context.Context, client *
 				if len(name) == len(dottedClusterDomain) {
 					continue
 				}
-				recordSetLogger := logger.WithField(
-					"recordset",
-					fmt.Sprintf("%s (%s)", aws.StringValue(recordSet.Name), aws.StringValue(recordSet.Type)),
-				)
 				// delete any matching record sets in the public hosted zone
 				if publicZoneID != "" {
-					if err := deleteMatchingRecordSetInPublicZone(ctx, client, publicZoneID, recordSet, logger); err != nil {
+					publicZoneLogger := logger.WithField("id", id)
+					if err := deleteMatchingRecordSetInPublicZone(ctx, client, publicZoneID, recordSet, publicZoneLogger); err != nil {
 						if lastError != nil {
-							logger.Debug(lastError)
+							publicZoneLogger.Debug(lastError)
 						}
 						lastError = errors.Wrapf(err, "deleting record set matching %#v from public zone %s", recordSet, publicZoneID)
 						// do not delete the record set in the private zone if the delete failed in the public zone;
 						// otherwise the record set in the public zone will get leaked
 						continue
 					}
+					recordSetLogger := logger.WithField("id", publicZoneID).WithField(
+						"recordset",
+						fmt.Sprintf("%s (%s)", aws.StringValue(recordSet.Name), aws.StringValue(recordSet.Type)),
```

Comment:
Extract this as a variable that can be shared by the two places were the "recordset" logger field is set, rather than duplicating this code. This will make the code a bit easier to read by being able to put the entire log event on a single line.
```
publicZoneLogger.WithField("recordset", recordsetValue).Debug("Deleted from public zone")
```

Or even use a `Fields` variable.
```
recordsetFields := logrus.Fields{"recordset": fmt.Sprintf("%s (%s)", aws.StringValue(recordSet.Name), aws.StringValue(recordSet.Type))}

publicZoneLogger.WithFields(recordsetFields).Debug("Deleted from public zone")
```